### ISSUE TITLE
Common code formatting style/apply JuliaFormatter by default?

### ISSUE THREAD

**Issue by meggart** [JuliaGeo/meta#16]
Hi all, 

I was wondering if we should, as a JuliaGeo organization, try to agree on a common code formatting style for the packages maintained under this umbrella. I have to admit that I am guilty of mixing different formatting styles in my packages (or not applying any guideline at all) and that this probably decreased readability of code. However, I recently looked into this and it looks like [JuliaFormatter](https://github.com/domluna/JuliaFormatter.jl) supports a variety of styles and provides [github workflows](https://github.com/julia-actions/julia-format) that either check PRs for compliance with the preferred style or regularly create formatting pull requests. 

I don't have a specific preference for one of the styles but before just picking one that I want to use for NetCDF.jl I was wondering if we can agree on one for the whole org (probably many of you already apply a certain style) and if we should install any of the above-mentioned actions for the JuliaGeo packages.

**Comment by visr** [comment]
I agree it would be nice to have a consistent style if possible, and JuliaFormatter is a great tool. So far I've used the default style and have been pretty happy with it. What style did you have in mind?

Regarding GitHub Actions, I am somewhat hesitant about the check for compliance. I've used that before on Python with `black`, and it can be a bit frustrating to see tests fail only because of a space or line break that is not perfect. Perhaps especially so for newcomers? Periodic PR could work I think.

**Comment by meggart** [comment]
> I agree it would be nice to have a consistent style if possible, and JuliaFormatter is a great tool. So far I've used the default style and have been pretty happy with it. What style did you have in mind?

No, I was experimenting with the default as well and happy with it, so if no one objects, I would use this one. 

> Regarding GitHub Actions, I am somewhat hesitant about the check for compliance. I've used that before on Python with black, and it can be a bit frustrating to see tests fail only because of a space or line break that is not perfect. Perhaps especially so for newcomers? Periodic PR could work I think.

This makes sense, I think merging these automatic formatting PRs should not be a ton of work (I think similar to the CompatHelper ones) and we do not shy away people by being overly restrictive with formatting.

**Comment by yeesian** [comment]
Yeah I've enjoyed my experience working with it, and have settled on https://github.com/yeesian/ArchGDAL.jl/blob/type-error-messages/.JuliaFormatter.toml (which is also consistent with https://github.com/jump-dev/JuMP.jl/blob/master/.JuliaFormatter.toml).

Happy to go with the default style for JuliaGeo packages that do not yet have a style guide.
